project/scripts/bisect.scala

```diff
@@ -1,24 +1,42 @@
-// Usage
-// > scala-cli project/scripts/dottyCompileBisect.scala -- [--run <main.class.name>] [<compiler-option> ...] <file1.scala> [<fileN.scala> ...]
-//
-// This script will bisect the compilation failure starting with a fast bisection on released nightly builds.
-// Then it will bisect the commits between the last nightly that worked and the first nightly that failed.
+/*
+This script will bisect a problem with the compiler based on success/failure of the validation script passed as an argument.
+It starts with a fast bisection on released nightly builds.
+Then it will bisect the commits between the last nightly that worked and the first nightly that failed.
+Look at the `usageMessage` below for more details.
+*/
 
 
 import sys.process._
 import scala.io.Source
 import Releases.Release
 import java.io.File
-import java.nio.file.{Files, Paths, StandardCopyOption}
+
+val usageMessage = """
+  |Usage:
+  |  > scala-cli project/scripts/bisect.scala -- <validation-script>
+  |
+  |The validation script should be executable and accept a single parameter, which will be the scala version to validate.
+  |Look at bisect-cli-example.sh and bisect-expect-example.exp for reference.
+  |Don't use the example scripts modified in place as they might disappear from the repo during a checkout.
```

Comment:
Can we create a temporary file for it like previously?

Reply:
I had this thought too but in the end this might introduce even more magic:
1) you modify an existing file in the repository
2) you run the bisect script
3) the modified file gets automagically copied to some random location
4) after checkouts during bisecting the modified file disappears from the repository
5) you realize that the result of bisecting makes no sense and you have to adjust the validating script to bisect again (that might not be a common case for simple compilation, but for bisecting REPL issues it was quite common from my experience)
6) you need to search for the temporary file with your script or write it again
Maybe this could be improved somehow but I'm not sure how to do it best and if this would be worth the effort